msgraph-sdk-raptor-compiler-lib/Models/IdentifierReplacer.cs

```diff
@@ -20,7 +20,7 @@ public class IdentifierReplacer
     /// regular expression to match strings like {namespace.type-id}
     /// also extracts namespace.type part separately so that we can use it as a lookup key.
     /// </summary>
-    private readonly Regex idRegex = new Regex(@"{([A-Za-z\.]+)\-id}", RegexOptions.Compiled);
+    private readonly Regex idRegex = new Regex(@"{([A-Za-z0-9\.]+)\-id}", RegexOptions.Compiled);

```

Comment:
```suggestion
    private readonly Regex idRegex = new Regex(@"{([\w\.]+)\-id}", RegexOptions.Compiled);

```

if underscore is ok as well.

Reply:
Underscore is not allowed by AGS explicitly in their naming guideline.

https://msazure.visualstudio.com/One/_wiki/wikis/Microsoft%20Graph%20Partners/103140/Naming-guidelines?anchor=general-guidelines